FSI/Stress.C

```diff
@@ -0,0 +1,142 @@
+#include "Stress.H"
+
+using namespace Foam;
+
+preciceAdapter::FSI::Stress::Stress
+(
+    const Foam::fvMesh& mesh,
+    const fileName& timeName,
+    const std::string solverType
+    /* TODO: We should add any required field names here.
+    /  They would need to be vector fields.
+    /  See FSI/Temperature.C for details.
+    */
+)
+:
+Force(mesh,solverType),
+mesh_(mesh),
+solverType_(solverType)
+{
+    //What about type "basic"?
+    if (solverType_.compare("incompressible") != 0 && solverType_.compare("compressible") != 0) 
+    {
+        FatalErrorInFunction
+            << "Forces calculation does only support "
+            << "compressible or incompressible solver type."
+            << exit(FatalError);
+    }
+    
+    dataType_ = vector;
+
+    Stress_ = new volVectorField
+    (
+        IOobject
+        (
+            "Stress",
+            timeName,
+            mesh,
+            IOobject::NO_READ,
+            IOobject::AUTO_WRITE
+        ),
+        mesh,
+        dimensionedVector
+        (
+            "pdim",
+            dimensionSet(1,-1,-2,0,0,0,0),
+            Foam::vector::zero
+        )
+    );
+}
+
+void preciceAdapter::FSI::Stress::write(double * buffer, bool /*meshConnectivity*/, const unsigned int dim)
+{
+    // Compute stress. See the Forces function object.
+
+    // Stress tensor boundary field
+    tmp<volSymmTensorField> tdevRhoReff = this->devRhoReff();
```

Comment:
I understand that by deriving from `Force`, you want to reduce duplication, although currently we do duplicate a lot. A good idea would be to further extract some of the duplicate parts into separate protected methods (both for `Force` and `Stress`).

Reply:
Taking a closer look at the code, I think the duplication is here quite minimal, since the important difference in the `write` function, namely the calculation using the normal vector, is included in every calculation step. Anyway, we could maybe unify the read (not supported) function. These are the only two methods, which are present here. 
I would propose we keep it like this at the moment. What do you think?

Reply:
This is also my feeling, let's keep it like this. :+1: